### PR TITLE
Make stable private modules public and adjust names

### PR DESCRIPTION
**Pull request by adrianeboyd**
## Description

* `spacy.ml._character_embed` -> `spacy.ml.character_embed`
* `spacy.tokens._serialize` -> `spacy.tokens.doc_bin`
* `spacy.tokens._retokenize` -> `spacy.tokens.retokenizer`
* `spacy.tokens._dict_proxies` -> `spacy.tokens.span_groups`

### Types of change

?

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
Maybe `retokenize` should be `retokenizer`?

File: setup.py [hunk 1]
```diff
@@ -60,7 +60,7 @@
     "spacy.tokens.span_group",
     "spacy.tokens.graph",
     "spacy.tokens.morphanalysis",
-    "spacy.tokens._retokenize",
+    "spacy.tokens.retokenize",
```
Comment (svlandeg) [inline setup.py:63]: `retokenizer` would definitely be fine by me!

**Review by svlandeg — COMMENTED** [review]
Oh god yes, I think these renamings are great!

**Comment by svlandeg** [comment]
https://github.com/explosion/spaCy/pull/10940 will remove `_precomputable_affine` so let's take it out of this refactor to avoid merge conflicts.

File: spacy/pipeline/attribute_ruler.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 from ..scorer import Scorer
 from ..symbols import IDS
 from ..tokens import Doc, Span
-from ..tokens._retokenize import normalize_token_attrs, set_token_attrs
+from ..tokens.retokenize import normalize_token_attrs, set_token_attrs
```
Comment (svlandeg) [inline spacy/pipeline/attribute_ruler.py:14]: There's a few more imports that need updating

**Pull request merged**